drivers/sensor/ite_vcmp_it8xxx2/vcmp_ite_it8xxx2.c

```diff
@@ -0,0 +1,398 @@
+			vcmp_enable(dev, 1);
+		} else {
+			vcmp_enable(dev, 0);
+			clear_vcmp_status(dev, config->vcmp_ch);
+		}
+
+		break;
+	default:
+		ret = -ENOTSUP;
+	}
+
+	return ret;
+}
+
+static int vcmp_ite_it8xxx2_attr_get(const struct device *dev,
+				     enum sensor_channel chan,
+				     enum sensor_attribute attr,
+				     struct sensor_value *val)
+{
+	return -ENOTSUP;
+}
+
+static int vcmp_ite_it8xxx2_trigger_set(const struct device *dev,
+					const struct sensor_trigger *trig,
+					sensor_trigger_handler_t handler)
+{
+	const struct vcmp_it8xxx2_config *const config = dev->config;
+	struct vcmp_it8xxx2_data *const data = dev->data;
+
+	if (trig->type != SENSOR_TRIG_THRESHOLD ||
+	    trig->chan != SENSOR_CHAN_VOLTAGE) {
+		return -ENOTSUP;
+	}
+
+	data->handler = handler;
+
+	vcmp_work_addr[config->vcmp_ch] = (uint32_t) &data->work;
+
+	return 0;
+}
+
+static int vcmp_it8xxx2_channel_get(const struct device *dev,
+				    enum sensor_channel chan,
+				    struct sensor_value *val)
+{
+	const struct vcmp_it8xxx2_config *const config = dev->config;
+
+	if (chan != SENSOR_CHAN_VOLTAGE) {
+		return -ENOTSUP;
+	}
+
+	/*
+	 * It8xxx2 adc and comparator module read automatically, according to
+	 * {ADCCTS1, ADCCTS2} and VCMPSCP register setting.
+	 */
+	val->val1 = config->vcmp_ch;
+
+	return 0;
+}
+
+/*
+ * All voltage comparator channels share one irq interrupt, so we
+ * need to handle all channels, when the interrupt fired.
+ */
+static void vcmp_it8xxx2_isr(const struct device *dev)
+{
+	const struct vcmp_it8xxx2_config *const config = dev->config;
+	volatile uint8_t *reg_vcmpsts = (uint8_t *)config->reg_vcmpsts;
+	volatile uint8_t *reg_vcmpsts2 = (uint8_t *)config->reg_vcmpsts2;
+	int idx, status;
+
+	/* Find out which voltage comparator triggered */
+	status = *reg_vcmpsts & VCMP_REG_MASK;
+	status |= (*reg_vcmpsts2 & VCMP_REG_MASK) << 3;
+
+	for (idx = VCMP_CHANNEL_0; idx < VCMP_CHANNEL_CNT; idx++) {
+		if (status & BIT(idx)) {
+			/* Trigger specific channel callback function in work queue */
+			if (vcmp_work_addr[idx]) {
+				k_work_submit((struct k_work *) vcmp_work_addr[idx]);
+			}
+			/* W/C voltage comparator specific channel interrupt status */
+			clear_vcmp_status(dev, idx);
+		}
+	}
+
+	/* W/C voltage comparator irq interrupt status */
+	ite_intc_isr_clear(config->irq);
+}
+
+static int vcmp_it8xxx2_init(const struct device *dev)
+{
+	const struct vcmp_it8xxx2_config *const config = dev->config;
+	struct vcmp_it8xxx2_data *const data = dev->data;
+	volatile uint8_t *reg_vcmpxctl = (uint8_t *)config->reg_vcmpxctl;
+	volatile uint8_t *reg_vcmpxcselm = (uint8_t *)config->reg_vcmpxcselm;
+	volatile uint8_t *reg_vcmpscp = (uint8_t *)config->reg_vcmpscp;
+
+	/* Disable voltage comparator specific channel before init */
+	vcmp_enable(dev, 0);
+
+	/*
+	 * ADC channel signal output to voltage comparator,
+	 * so we need to set ADC channel to alternate mode first.
+	 */
+	adc_channel_setup(config->adc, &data->adc_ch_cfg);
```

Comment:
you need to check for `adc` device readiness

Reply:
Vcmp module only need to set the pin to adc alt mode, then adc HW will automatically and periodically scan the pin volt base on adc init IT8XXX2_ADC_ADCCTS[0,1]  register setting (default period is longer), so we needn't to call adc_read() api. And adc init priority (CONFIG_ADC_INIT_PRIORITY  = 50) is higher than vcmp init priority (CONFIG_SENSOR_INIT_PRIORITY = 90), so does it need to check  adc readiness?

Reply:
You're using the ADC device (via the ADC API) without checking for readiness. If device is not ready, it means `adc` driver instance has not gone through its init call.

Reply:
Ack, but how to check adc init() finish or not?

Reply:
you can use `device_is_ready()` for that. If a device driver completes initialization successfully (init returns 0) it will be flagged as ready.

Reply:
Done